ecpy_hqc_legacy/instruments/drivers/visa/yokogawa.py

```diff
@@ -175,20 +252,25 @@ def output(self, value):
         """
         on = re.compile('on', re.IGNORECASE)
         off = re.compile('off', re.IGNORECASE)
-        if on.match(value) or value == 1:
-            self.write(':OUTPUT ON')
-            if self.ask(':OUTPUT?') != '1':
-                raise InstrIOError(cleandoc('''Instrument did not set correctly
-                                            the output'''))
-        elif off.match(value) or value == 0:
-            self.write(':OUTPUT OFF')
-            if self.ask(':OUTPUT?') != '0':
-                raise InstrIOError(cleandoc('''Instrument did not set correctly
-                                            the output'''))
+        if self.voltage == 0:
```

Comment:
Perhaps add a comment about why checking the voltage is enough (it was not immediately obvious to me, so may be worth it)

Reply:
Ok, added this:        
# output is only switched on if voltage or current is 0
# note that self.voltage gets the source level, which is the voltage
# in voltage mode, and the current in current mode.